### DISCUSSION TITLE
Coupling two weak forms

### DISCUSSION THREAD

**Discussion by Hasupama** [firedrakeproject/firedrake#3301] [Firedrake support]
Hi,
I want to solve Poisson equation and continuity equation simultaneously. Weak forms are,

`Weak_form_poisson = x[0]*( eps*phi.dx(i)*del_phi.dx(i) - e*(-Ne*del_phi )*dx`
`Weak_form_continuity= x[0]*( Ne-Ne0/Dt)*del_Ne  - source*del_Ne )*dx`

When I tried to solve these two together(`Weak_form_poisson+Weak_form_continuity=0`), It gives an error _DIVERGED_FNORM_NAN_

Is it incorrect to solve the addition of two weak forms?

Appreciate your suggestions on this problem. Thanks.

**Comment by connorjward** [comment]
Does it work if you solve for just one of those forms?

**Comment by dham** [comment]
There's really not enough information here to answer. The error indicates that the solver diverged, which is probably because the problem was ill-posed. That could either be because of a maths issue or because of a mistake translating that maths into code. Please post a minimal failing example (i.e. the shortest complete piece of code that illustrates the issue).

**Comment by colinjcotter** [comment]
What is x[0] ?

________________________________________

**Reply by dham** [reply]
What are any of the symbols? That's why an MFE is needed.
